src/Components/Notifications/NotificationsList.tsx

```diff
@@ -240,10 +244,18 @@ export default function NotificationsList({
 
   const handleSubscribeClick = () => {
     const status = isSubscribed;
-    if (status === "NotSubscribed" || status === "SubscribedOnAnotherDevice") {
-      subscribe();
-    } else {
-      unsubscribe();
+    if (
+      navigator.serviceWorker !== null &&
+      navigator.serviceWorker !== undefined
+    ) {
+      if (
+        status === "NotSubscribed" ||
+        status === "SubscribedOnAnotherDevice"
+      ) {
+        subscribe();
+      } else {
+        unsubscribe();
+      }
```

Comment:
This is a bit hard to follow, let's re-write it 

```suggestion
    if (!navigator.serviceWorker) {
        return;
    }
      
    if (["NotSubscribed", "SubscribedOnAnotherDevice"].includes(status)) {
        subscribe();
    } else {
        unsubscribe();
     }
```